std/conv.d

```diff
@@ -1883,6 +1883,47 @@ if (isInputRange!S && !isInfinite!S && isSomeChar!(ElementEncodingType!S) &&
     assert(to!int(byDchar("10"), 10) == 10);
 }
 
+/**
+String, or string-like input range, to char type not directly
+supported by parse parses the first dchar of the source.
+
+Returns: the first code point of the input range, converted
+         to type T.
+
+Throws: ConvException if the input range contains more than
+        a single code point, or if the code point does not
+        fit into a code unit of type T.
+*/
+private T toImpl(T, S)(S value)
+if (isSomeChar!T && !is(typeof(parse!T(value))) &&
+    is(typeof(parse!double(value))))
+{
+    immutable dchar codepoint = parse!dchar(value);
+    if (!value.empty)
+        throw new ConvException(convFormat("Cannot convert \"%s\" to %s because it " ~
+                                           "contains more than a single code point.",
+                                           value, T.stringof));
+    auto equivalentTArray = to!(T[])(codepoint);
+    if (equivalentTArray.length > 1)
+        throw new ConvException(convFormat("Code point '%s' does not fit into a single " ~
+                                           "%s code unit", codepoint, T.stringof));
```

Comment:
Slightly adjust the message here, remember the input was a string, not a code point.

Reply:
Updated, let's see if it's better now.